WordPress/Classes/Services/AccountService.m

```diff
@@ -35,22 +35,20 @@ @implementation AccountService
  */
 - (WPAccount *)defaultWordPressComAccount
 {
-    NSString *uuid = [[NSUserDefaults standardUserDefaults] stringForKey:DefaultDotcomAccountUUIDDefaultsKey];
-    if (uuid.length == 0) {
+    NSURL *myObjectURL = [NSURL URLWithString:[[NSUserDefaults standardUserDefaults] objectForKey:DefaultDotcomAccountIDDefaultsKey]];
+    NSManagedObjectID *myObjectID = [self.managedObjectContext.persistentStoreCoordinator managedObjectIDForURIRepresentation:myObjectURL];
+
+    if (!myObjectID) {
         return nil;
     }
 
-    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Account"];
-    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"uuid == %@", uuid];
-    fetchRequest.predicate = predicate;
-    
     NSError *error = nil;
-    NSArray *fetchedObjects = [self.managedObjectContext executeFetchRequest:fetchRequest error:&error];
+    WPAccount *foundAccount = [self.managedObjectContext existingObjectWithID:myObjectID error:&error];
```

Comment:
I think we could make the `userID` work by ensuring we have all the details beforehand, although I'd postpone the creation of the `WPAccount` object, not just the default account change.

However, I think that might require some more discussion and it's not really required for multiple account support, so I'd revert to UUID and handle that on a separate PR.